File: clients/kotlin_client/client/src/main/kotlin/core/Strategy.kt
package core

import core.API.Elevator
import core.API.Passenger

class Strategy : BaseStrategy() {
  override fun onTick(myPassengers: List<Passenger>, myElevators: List<Elevator>, enemyPassengers: List<Passenger>, enemyElevators: List<Elevator>) {
    for (e in myElevators) {
      for (p in myPassengers) {
        if (p.state < 5) {
          if (e.state != 1) {
            e.goToFloor(p.fromFloor)
          }
          if (e.floor == p.fromFloor) {
            p.setElevator(e)
          }
        }
      }
      if (e.passengers.size > 0 && e.state != 1) {
        e.goToFloor(e.passengers[0].destFloor)
      }
    }
  }
}

File: clients/go_client/client/Strategy.go
package main

import (
	"github.com/pomkac/aicups/clients/go_client/client/core/API"
)

type Strategy struct {
	API.BaseStrategy
}

// Вызывается при инициализации клиента
func (s *Strategy) Init(){

}

func (s *Strategy) OnTick(myPassengers []*API.Passenger, myElevators []*API.Elevator,
				enemyPassengers []*API.Passenger, enemyElevators []*API.Elevator) error {

	for i := 0; i < len(myElevators); i++ {
		elevator := myElevators[i]

		for j := 0; j < len(myPassengers); j++ {
			passenger := myPassengers[j]

			if passenger.State() < 5{
				if elevator.State() != 1{
					elevator.GoToFloor(passenger.FromFloor())
				}
				if elevator.Floor() == passenger.Floor(){
					passenger.SetElevator(elevator)
				}
			}
		}

		if len(elevator.Passengers()) > 0 && elevator.State() != 1{
			elevator.GoToFloor(elevator.Passengers()[0].DestFloor())
		}
	}

	return nil
}

